tests/rule_based_profiler/test_rule_based_profiler.py

```diff
@@ -0,0 +1,399 @@
+from typing import Any, Dict, List, Optional
+
+from great_expectations.rule_based_profiler.rule import Rule
+from great_expectations.rule_based_profiler.types import ParameterContainer
+
+
+def test_reconcile_profiler_variables_no_overrides(
+    profiler_with_placeholder_args,
+    variables_multi_part_name_parameter_container: ParameterContainer,
+):
+    variables: Dict[str, Any] = {}
+    effective_variables: Optional[
+        ParameterContainer
+    ] = profiler_with_placeholder_args.reconcile_profiler_variables(variables=variables)
+    assert effective_variables == variables_multi_part_name_parameter_container
+
+
+def test_reconcile_profiler_variables_with_overrides(
+    profiler_with_placeholder_args,
+):
+    variables: Dict[str, Any] = {
+        "false_positive_threshold": 2.0e-2,
+        "sampling_method": "bootstrap",
+        "mostly": 8.0e-1,
+    }
+    effective_variables: Optional[
+        ParameterContainer
+    ] = profiler_with_placeholder_args.reconcile_profiler_variables(variables=variables)
+    assert effective_variables.to_dict()["parameter_nodes"]["variables"][
+        "variables"
+    ] == {
+        "false_positive_threshold": 2.0e-2,
+        "sampling_method": "bootstrap",
+        "mostly": 8.0e-1,
+    }
+
+
+def test_reconcile_profiler_rules_no_overrides(
+    profiler_with_placeholder_args,
+):
+    rules: Dict[str, Dict[str, Any]] = {}
+    effective_rules: List[
+        Rule
+    ] = profiler_with_placeholder_args.reconcile_profiler_rules(rules=rules)
+    assert effective_rules == profiler_with_placeholder_args.rules
+
+
+def test_reconcile_profiler_rules_new_rule_override(
+    profiler_with_placeholder_args,
+):
+    rules: Dict[str, Dict[str, Any]] = {
+        "rule_0": {
+            "domain_builder": {
+                "class_name": "ColumnDomainBuilder",
+                "module_name": "great_expectations.rule_based_profiler.domain_builder",
+            },
+            "parameter_builders": [
+                {
+                    "class_name": "MetricMultiBatchParameterBuilder",
+                    "module_name": "great_expectations.rule_based_profiler.parameter_builder",
+                    "name": "my_parameter",
+                    "metric_name": "my_metric",
+                },
+                {
+                    "class_name": "NumericMetricRangeMultiBatchParameterBuilder",
+                    "module_name": "great_expectations.rule_based_profiler.parameter_builder",
+                    "name": "my_other_parameter",
+                    "metric_name": "my_other_metric",
```

Comment:
Can this go in a fixture? This single test feels very verbose and we can possibly reuse the data herein.

Reply:
@cdkini This is a one-time example (not sure if we need this for any other purposes).  Thanks.